lib/src/room.dart

```diff
@@ -270,6 +279,59 @@ class KurentoRoom extends ChangeNotifier implements Room {
     await _updateParticipantStatus();
   }
 
+  @override
+  Future<void> updateLocation(Position position) async {
+    if (!_mq.isConnected) {
+      _log.warning('MQTT client is not connected, cannot send gps coordinates');
+      return;
+    }
+    if (ServiceRequestState.started != _serviceRequestState) {
+      _log.warning('ServiceRequest is not started yet, not sending position ($_serviceRequestState)');
+      return;
+    }
+
+    List<Map<String, dynamic>> data = [
+      {'instrumentationType': 'TYPE_GPS',
+        'paramName': 'LAT',
+        'paramValue': position.latitude},
+      {'instrumentationType': 'TYPE_GPS',
+        'paramName': 'LONG',
+        'paramValue': position.longitude},
+      {'instrumentationType': 'TYPE_GPS',
+        'paramName': 'BEARING',
+        'paramValue': position.heading},
+      {'instrumentationType': 'TYPE_GPS',
+        'paramName': 'HORIZONTAL_ACCURACY',
+        'paramValue': position.accuracy},
+      if (null != position.headingAccuracy) {'instrumentationType': 'TYPE_GPS',
+        'paramName': 'DIRECTION_ACCURACY',
+        'paramValue': position.headingAccuracy},
+    ];
+
+    Map<String, dynamic> payload = {
+      'serviceRequestID': _serviceRequest.id,
+      'agentid': _agentId ?? 0,
+      'data': data
+    };
+
+    Map<String, dynamic> data2 = {
+      'userId': _serviceRequest.userId,
+      'lt': position.latitude,
+      'lg': position.longitude,
+    };
+
+    try {
+      _log.info('Publish location info');
+      _log.finest(payload);
+      await Future.wait([
+        _mq.publish(_siPubTopic, MqttQos.atMostOnce, jsonEncode(payload)),
+        _mq.publish(_gpsLocationTopic, MqttQos.atMostOnce, jsonEncode(data2)),
+      ]);
+    } catch (e) {
+      _log.warning('Unable to send data to topic $_siPubTopic & $_gpsLocationTopic because of $e');
```

Comment:
When logging an error, should we pass `e` in as a parameter instead of including it in the string?